File: agent/src/core/agent_state.py
# -*- coding: utf-8 -*-
"""
Defines the possible operational states of the agent.
"""
from enum import Enum, auto

class AgentState(Enum):
    """Enumeration of agent operational states."""
    STARTING = auto()         # Initial state during startup and initialization.
    IDLE = auto()             # Agent is running normally, connected (or attempting reconnect), waiting for commands.
    FORCE_RESTARTING = auto() # State when handling --force restart via IPC
    UPDATING_STARTING = auto() # Update process initiated
    UPDATING_DOWNLOADING = auto() # Downloading update package
    UPDATING_VERIFYING = auto() # Verifying package integrity/signature
    UPDATING_EXTRACTING_UPDATER = auto() # Extracting updater tool
    UPDATING_PREPARING_SHUTDOWN = auto() # Preparing to hand off to updater
    SHUTTING_DOWN = auto()    # Agent is in the process of stopping gracefully.
    STOPPED = auto()          # Agent has fully stopped (final state, not typically set internally).

File: agent/CMSAgent/src/Core/AgentState.cs
namespace CMSAgent.Core
{
    /// <summary>
    /// Represents the current state of the agent
    /// </summary>
    public enum AgentState
    {
        /// <summary>
        /// Agent has not been started yet
        /// </summary>
        NotStarted,
        
        /// <summary>
        /// Agent is in the process of starting
        /// </summary>
        Starting,
        
        /// <summary>
        /// Agent is running normally
        /// </summary>
        Running,
        
        /// <summary>
        /// Agent is in the process of stopping
        /// </summary>
        Stopping,
        
        /// <summary>
        /// Agent has stopped
        /// </summary>
        Stopped,
        
        /// <summary>
        /// Agent encountered an error
        /// </summary>
        Error
    }
}